OasisDEX/oasis-borrow
# FEAT - 🧭 Add status and activity handlers

# [Title](https://shortcutLink)

<please insert a shortcut link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>
- item 1
  
## How to test 🧪
Seed the database and check the disscover page.
Piotr.
```
import { prisma, PrismaClient } from '@prisma/client';
import { Discover, Prisma } from '@prisma/client';
import { timestamp } from 'rxjs';
const prismaClient = new PrismaClient();

function getRandomInt(max: number) {
  return Math.floor(Math.random() * max);
}

enum Protocols {
  AAVE = 'AAVE',
  MAKER = 'MAKER',
  COMPOUND = 'COMPOUND',
}

enum Assets {
  ETH = 'ETH',
  WBTC = 'WBTC',
  UNI_LP = 'UNI_LP',
  LINK = 'UNI_LP',
  MANA = 'UNI_LP',
  MATIC = 'MATIC',
  GUSD = 'GUSD',
  CURVE_LP = 'CURVE_LP',
  YFI = 'YFI',
}
const statuses = [
  {
    kind: 4,
    additionalData: {
      toStopLoss: 12.61,
    },
  },

  {
    kind: 3,
    additionalData: {
      tillLiquidation: 1.02,
    },
  },

  { kind: 2 },
  { kind: 1 },
];
const activities = [
  {
    kind: 3,
    additionalData: {
      daysAgo: 2,
      timestamp: Date.now().toString(),
    },
  },
  {
    kind: 2,
    additionalData: {
      daysAgo: 3,
      timestamp: Date.now().toString(),
    },
  },
  {
    kind: 1,
    additionalData: {
      daysAgo: 1,
      timestamp: Date.now().toString(),
    },
  },
  {
    kind: 6,
    additionalData: {
      daysAgo: 9,
      timestamp: Date.now().toString(),
    },
  },
];

const main = async () => {
  const data: Discover[] = [];

  Object.keys(Protocols).forEach(async (p) => {
    for (let i = 1; i < 1000; i++) {
      const r = {
        protocol_id: p,
        position_id: i.toString(),
        collateral_type:
          Object.keys(Assets)[
            Math.floor(Math.random() * (Object.keys(Assets).length / 2))
          ],
        vault_collateral: new Prisma.Decimal(Math.random() * 1000),
        vault_debt: new Prisma.Decimal(Math.random() * 10000),
        vault_normalized_debt: new Prisma.Decimal(Math.random() * 10000),
        yield_30d: new Prisma.Decimal(Math.random() * 10),
        pnl_all: new Prisma.Decimal(Math.random() * 1000000),
        pnl_ytd: new Prisma.Decimal(Math.random() * 1000000),
        pnl_1d: new Prisma.Decimal(Math.random() * 1000000),
        pnl_7d: new Prisma.Decimal(Math.random() * 1000000),
        pnl_30d: new Prisma.Decimal(Math.random() * 1000000),
        pnl_365d: new Prisma.Decimal(Math.random() * 1000000),
        status: statuses[getRandomInt(3)],
        last_action: activities[getRandomInt(3)],
        createdAt: new Date(),
        updatedAt: new Date(),
      };

      data.push(r);
    }
  });
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment

  Object.keys(Assets).forEach(async (k) => {
    await prismaClient.collateralType.create({
      data: {
        collateral_name: k,
        token: k,
        current_price: new Prisma.Decimal(Math.random() * 1000000),
        next_price: new Prisma.Decimal(Math.random() * 1000000),
        market_price: new Prisma.Decimal(Math.random() * 1000000),
        liquidation_ratio: new Prisma.Decimal(Math.random() * 10),
        liquidation_penalty: new Prisma.Decimal(1.13),
        rate: new Prisma.Decimal(Math.random() * 1),
      },
    });
  });
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  await prismaClient.discover.createMany({ data });
};
main()
  .catch((e) => {
    console.error(e);
    process.exit(1);
  })
  .finally(async () => {
    // close Prisma Client at the end
    await prismaClient.$disconnect();
  });
```
